Classes/TYPO3/TYPO3CR/Search/Indexer/NodeIndexingManager.php

```diff
@@ -135,4 +149,16 @@ public function flushQueues() {
 			$flush();
 		}
 	}
+
+	/**
+	 * The callback is executed with Indexing disabled
+	 *
+	 * @param callable $callback
+	 */
+	public function withoutIndexing(callable $callback)
+	{
+		$this->enabled = false;
+		$callback();
+		$this->enabled = true;
+	}
```

Comment:
any reason to use a callable instead of a closure? also there's no handling of exceptions, so the indexer could be disabled if one occurs and is caught..

see https://github.com/neos/flow/blob/master/Classes/TYPO3/Flow/Security/Context.php#L220